src/main/java/com/gooddata/http/client/GoodDataHttpClient.java

```diff
@@ -273,9 +273,14 @@ public HttpResponse execute(HttpUriRequest request, HttpContext context) throws
     @Override
     public HttpResponse execute(HttpHost target, HttpRequest request, HttpContext context) throws IOException {
         if (context == null) {
-            context = this.context;
+            context = this.defaultContext;
+        } else if (context != this.defaultContext) {
```

Comment:
is it ok to compare contexts by instance equality when there is new instance created and this method is called recursively?